IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -250,6 +275,21 @@ - (void)saveTelemetryToDisk
     {
         NSData *dataToArchive = [NSKeyedArchiver msidArchivedDataWithRootObject:self requiringSecureCoding:YES error:nil];
         
+        // If data is larger than 10x the 4kB limit, start removing errors from the telemetry
+        // object until it fits under the limit. Limit is 10x to allow for additional metadata
+        // that NSKeyedArchiver adds to the archive
+        if (dataToArchive.length > [MSIDLastRequestTelemetry telemetryStringSizeLimit] * 10)
+        {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, nil, @"Telemetry size over limit when saving to disk, cutting down to limit", nil);
+            
+            while (_errorsInfo.count > 0 &&
+                   [NSKeyedArchiver msidArchivedDataWithRootObject:self requiringSecureCoding:YES error:nil].length > [MSIDLastRequestTelemetry telemetryStringSizeLimit] * 10)
```

Comment:
This looks a little inefficient. Is there any approximation we could have? (e.g. number of errors permitted that always fits the size). Alternatively, could we just write up to the limit of data to disk and deserialization next time will take care of cleaning it up?

Reply:
I like the idea of using number of errors, I had that idea as well. The reason I didn't go with it initially was because I wanted to see if there was a way to connect the server limit for serialized string size to the limit on disk archive size so any future updates to these sizes would be made in only one place. After some thought, these two sizes are too separate to combine in this way, especially because archiving adds various metadata so the size of the serialized string and the archive size aren't as obviously related.

After some testing, around 75 errors with minimal error descriptions corresponds to about 8kB, so that's where I'll set the limit.